### PR TITLE
Eng 12920 change sph on the fly

### PR DESCRIPTION
**Pull request by GuaGuaDean**
Added exception handling when detecting site per host change in deployment file

### CODE REVIEW

File: src/frontend/org/voltdb/sysprocs/UpdateApplicationBase.java [hunk 1]
```diff
@@ -107,6 +107,9 @@ public static CatalogChangeResult prepareApplicationCatalogDiff(String invocatio
         try {
             // catalog change specific boiler plate
             CatalogContext context = VoltDB.instance().getCatalogContext();
+            if (context.getDeployment().getCluster().getSitesperhost() != CatalogUtil.parseDeploymentFromString(operationString).getCluster().getSitesperhost()) {
+            	throw new Exception("You are trying to change the site per host.");
```
Comment (zeemanhe) [inline src/frontend/org/voltdb/sysprocs/UpdateApplicationBase.java]: CatalogUtil.parseDeploymentFromString(operationString).getCluster() may throw a null pointer.  suggestion:
move   DeploymentType dt  = CatalogUtil.parseDeploymentFromString(deploymentString);
            if (dt == null) {
                retval.errorMsg = "Unable to update deployment configuration: Error parsing deployment string";
                return retval;
            }
before the validation

File: src/frontend/org/voltdb/sysprocs/UpdateApplicationBase.java [hunk 2]
```diff
@@ -228,6 +231,7 @@ else if ("@AdHoc".equals(invocationName)) {
             // because the catalog bytes does not contain any deployments but only schema related contents
             // the command log reply needs it to generate a correct catalog diff
             DeploymentType dt  = CatalogUtil.parseDeploymentFromString(deploymentString);
+            Thread.dumpStack();
```
Comment (zeemanhe) [inline src/frontend/org/voltdb/sysprocs/UpdateApplicationBase.java]: remove Thread.dumpStack() and move these as suggested in the above comment.

File: src/frontend/org/voltdb/sysprocs/UpdateApplicationBase.java (hunk 1 above)
Comment (zeemanhe) [reply src/frontend/org/voltdb/sysprocs/UpdateApplicationBase.java]: do not throw Exception here.  set the error message in the retval instead.
retval.errorMsg = "Unable to update deployment configuration: the site per host can not be updated";

Comment (GuaGuaDean) [reply src/frontend/org/voltdb/sysprocs/UpdateApplicationBase.java]: Corrected.

File: src/frontend/org/voltdb/sysprocs/UpdateApplicationBase.java (hunk 2 above)
Comment (GuaGuaDean) [reply src/frontend/org/voltdb/sysprocs/UpdateApplicationBase.java]: Corrected.

**Pull request merged**